### PR TITLE
hw: mcu: kinetis: add MCGIRCLK configuration

### PR DESCRIPTION
**Pull request by utzig**
Allow setting the speed and divider for MCGIRCLK.

### CODE REVIEW

**Review by kasjer — COMMENTED** [review]
Original code used different settings for mcgConfig_BOARD_xxxx
Now one settings (from syscfg) is used in all places maybe it is OK, I don't know.

File: hw/mcu/nxp/kinetis/syscfg.yml [hunk 1]
```diff
@@ -24,6 +24,17 @@ syscfg.defs:
...
         restriction: $notnull
 
+    KINETIS_USE_FAST_MCGIRCLK:
+        description: >
+            Use fast clock for MCGIRCLK. When enabled MCGIRCLK will clock
+            at 4MHz; when disabled 32768Hz.
+        value: 0
+    KINETIS_MCGIRCLK_FCRDIV:
+        description: >
+            Divider for MCGIRCLK. The divider used equals: value << 1.
```
[the remark this answers is not in the document]
Comment (utzig) [reply hw/mcu/nxp/kinetis/syscfg.yml:34]: Yep, good catch!

**Comment by utzig** [comment]
> Original code used different settings for mcgConfig_BOARD_xxxx
> Now one settings (from syscfg) is used in all places maybe it is OK, I don't know.

Yeah, you got a point. It doesn't break anything apart from having a different configuration; only one of those is ever used, but there are more because this file is basically "auto-generated" and supports different running modes (high-speed, low-power). It would be better to remove that code and just do the clock configuration based on syscfgs for the changing fields; I'll take a look.

**Review by kasjer — APPROVED** [review]
LGTM

**Comment by sjanc** [comment]
Fabio, is this something that you want to be merged for release?

**Comment by utzig** [comment]
> Fabio, is this something that you want to be merged for release?

No, I'd rather not change the configuration defaults that already exist, so this needs some extra work, which I am currently unable to do, so I'll close it until it can be resumed.

**Pull request closed without merging**